test/end2end_test.go

```diff
@@ -3901,6 +3901,36 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+func testClientInvalidStreamID(t *testing.T, e env) {
```

Comment:
Looking at the tests, we are missing a test for the case where the server sends an invalid stream ID as well. Might be worth adding that add that as part of this diff.

Reply:
Looking at the behavior of servertester.go, we send requests to server to test the server and server sends appropriate response to the client conn? 

If I want to test if the client is throwing error if it receives odd stream IDs from the server, servertester is the not the right one to use IIUC?

Reply:
That's correct. Your best bet must be a similar [clienttester](https://github.com/AnandInguva/grpc-go/blob/master/test/clienttester.go) in the same test package.

Reply:
As discussed offline, since the stream is always initiated by the client, we don't need to test the stream id on the client side.